src/websockethandler.cpp

```diff
@@ -15,6 +15,49 @@ namespace {
 Logger logger(LOG_MAIN, "WebSocketHandler");
 }  // namespace
 
+ExponentialBackoffStrategy::ExponentialBackoffStrategy() {
+  MVPN_COUNT_CTOR(ExponentialBackoffStrategy);
+
+  connect(&m_retryTimer, &QTimer::timeout, this,
+          &ExponentialBackoffStrategy::executeNextAttempt);
+  m_retryTimer.setSingleShot(true);
+}
+
+#ifdef UNIT_TEST
+void ExponentialBackoffStrategy::testOverrideBaseRetryInterval(
+    int newInterval) {
+  m_baseInterval = newInterval;
+}
+
+void ExponentialBackoffStrategy::testOverrideMaxRetryInterval(int newInterval) {
+  m_maxInterval = newInterval;
+}
+#endif
+
+/**
+ * @brief Schedules the next attempt to execute a given function.
+ *
+ * Everytime a new attempt is scheduled the interval will be exponentially
+ * larger.
+ */
+int ExponentialBackoffStrategy::scheduleNextAttempt() {
+  int retryInterval = qPow(m_baseInterval, m_retryCounter);
+  if (retryInterval < m_maxInterval) {
```

Comment:
I have a question here. `m_baseinterval` is 5000. `m_maxInterval` is 3600000. So, the interval will be only 5000.
Because 5000^2 is 25000000 which is greater than 3600000.
I wonder if we should multiply `* 1000` later. In this way the intervals will be: [ 5, 25, 125, 625, 3125 ].
Am I right?

Reply:
!!! Yeap, you are totally right. I missed this in tests because I am overwriting the interval over there to 5...